### ISSUE TITLE
3.9 beta1+2/RC1: Structure Panel empty on Safari 15

### ISSUE THREAD

**Issue by sadrian80** [soflyy/oxygen-bugs-and-features#2505]
**Description:**
Using any version of the 3.9 release, the Structure Panel stays empty on initial load and shows upon reload of the window. I can replicate the issue on at least three machines using an actual release of Safari 15 on macOS 11 and 12. One machine is freshly installed yesterday via Internet recovery after an SSD swap. Installation on all machines is stock with no extensions installed.

Firefox Dev and Chromium don't show this issue.

**Sandbox:**
http://ambitious-mink.w5.wpsandbox.pro?upass=D968uS6eXGev

**Steps to reproduce the behavior:**
1. Edit template with Oxygen using Safari 15.
2. Open Structure Panel

![Screenshot 2021-11-05 at 21 19 29](https://user-images.githubusercontent.com/77888082/140575057-defc4eb1-066b-491b-b6e3-c3c06dd8e154.png)

**Comment by Spellhammer** [comment]
Unable to reproduce on Safari version 15.0 (16612.1.29.41.4, 16612) on the provided Sandbox install.

Are there any particular steps I need to take to see the problem? It's known that the Structure Pane's styling may be off on the first load after upgrading to 3.9, but we've never had a report of it being empty.

If you can provide more details that would be greatly appreciated.

**Comment by sadrian80** [comment]
I'll gladly fill in as much as I can. I can still reproduce the issue, even on a friends machine with no connection to me or my network. Also on an older Mac Mini running 10.3 with Safari 15 installed. While trying to reproduce I gained some new insights:

I did this screen recording on my Mac Pro. The machine is freshly installed and besides Creative Cloud, MS Office, Nova, Transmit and Little Snitch the system is 100% stock. I completely disabled Little Snitch, VPN and even the firewall in my router to have nothing interfere.

Here's the screen recording showing the issue in a new sandbox, quite boring to be honest: https://www.dropbox.com/s/n8cyawwcgh18hp8/oxygen_screencast.mp4?dl=0

Now, while exporting the recording I left the sandbox window opened. First thing I noticed was the fans on the Mac Pro spinning up, something I rarely had in years. Next the browser window shows the "This webpage is using significant energy." notice, followed by the structure panel appearing shortly after getting completely unresponsive at the same time.
![Screenshot 2021-11-07 at 15 29 49](https://user-images.githubusercontent.com/77888082/140650127-7c9b6c7b-57a0-4964-8952-02bb7a1b69d2.png)

I tried looking at the console which showed a stream of errors flowing in for angular.js
![Screenshot 2021-11-07 at 15 30 51](https://user-images.githubusercontent.com/77888082/140650120-7d7687a3-a3e1-4cbe-a59a-e960ee36931d.png)
:
![Screenshot 2021-11-07 at 15 56 05](https://user-images.githubusercontent.com/77888082/140650231-d967a0f2-19ed-44bd-91d0-fc3e51d19cad.png)

After some minutes the errors stopped, the window got responsive again. AND: now the issue is gone BUT only for that particular sandbox. I created yet another sandbox and I can replicate again.

This is a strange issue. 🧐

**Comment by freeskateonly** [comment]
I have same problems with stucture panel too – with Firefox 95.0b
![Снимок экрана 2021-11-08 в 12 06 43](https://user-images.githubusercontent.com/15000057/140706167-0c070b38-f36e-41b6-bdcd-6410910b434d.jpg)
3

**Comment by sadrian80** [comment]
> I have same problems with stucture panel too – with Firefox 95.0b
> ![Снимок экрана 2021-11-08 в 12 06 43](https://user-images.githubusercontent.com/15000057/140706167-0c070b38-f36e-41b6-bdcd-6410910b434d.jpg)
> 3

This looks like you have some additional plugin installed. It might not work with the Beta/RC release yet.

**Comment by KittenCodes** [comment]
@freeskateonly 

> I have same problems with stucture panel too – with Firefox 95.0b

That's usually caused by cached content. Please hard-refresh the builder.

**Comment by freeskateonly** [comment]
Thank you for advice. I can't find where exactly cached builder content that i need to flash.. Do u now? =)

**Comment by KittenCodes** [comment]
> Thank you for advice. I can't find where exactly cached builder content that i need to flash.. Do u now? =)

On Windows, you would need to reload the builder using Crtl + F5. I believe on Mac, it's Shift + Command + R.

**Comment by devilsinkpot** [comment]
> > Thank you for advice. I can't find where exactly cached builder content that i need to flash.. Do u now? =)
> 
> 
> 
> On Windows, you would need to reload the builder using Crtl + F5. I believe on Mac, it's Shift + Command + R.

CMD + ALT + R is reload from source.
CMD + ALT + E is flush cache for actual URL.

**Comment by wplit** [comment]
I've come across the 'Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!' error with the RC1 also

![iteractionsreached](https://user-images.githubusercontent.com/43051571/140918244-add7b63b-f6b3-455d-b5e8-ab4d852fb919.jpg)

**Comment by KittenCodes** [comment]
@wplit 

> I've come across the 'Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!' error with the RC1 also

This also occurs in v3.8.1, it's usually caused by something in Safari v15 - we're currently investigating. Is that the browser you're using?

**Comment by wplit** [comment]
@KittenCodes No, this is in Chrome on Mac. It's been happening when I've been copying elements from an old version of Oxygen to the new RC1. (copying using Hydrogen plugin) 

I'll try to find a reliable way to recreate it.

**Comment by Cr3epy** [comment]
Same issue with the 3.9 Version in Safari and the latest MAcOS. It does work in Chrome it seems Safari is blocking something in the Structure panel

**Comment by Spellhammer** [comment]
I believe this has been fixed for a while. Closing.
